### ISSUE TITLE
Calculation of AUC metric

### ISSUE THREAD

**Issue by weilonghu** [hwwang55/GCNRec#22]
```
def ctr_eval(sess, model, data, batch_size):
    start = 0
    auc_list = []
    f1_list = []
    while start + batch_size <= data.shape[0]:
        auc, f1 = model.eval(sess, get_feed_dict(model, data, start, start + batch_size))
        auc_list.append(auc)
        f1_list.append(f1)
        start += batch_size
    return float(np.mean(auc_list)), float(np.mean(f1_list))
```
In this function, you calculate AUC in every batch and take their average value as the final AUC.  But as far as I know AUC needs to be globally sorted in the test set.  Can you explain it ?
